src/fu-engine.c

```diff
@@ -2672,10 +2675,14 @@ fu_engine_emulation_load(FuEngine *self, GInputStream *stream, GError **error)
 	if (!fu_engine_emulation_load_json_blob(self, json_blob, error))
 		return FALSE;
 
-	/* load archive */
-	archive = fu_archive_new_stream(stream, FU_ARCHIVE_FLAG_NONE, error);
-	if (archive == NULL)
-		return FALSE;
+	/* may have directly received a phase */
+	if (!g_str_has_suffix(filename, ".json")) {
+		/* load archive */
+		archive = fu_archive_new_stream(stream, FU_ARCHIVE_FLAG_NONE, error);
+		if (archive == NULL)
+			return FALSE;
+	}
+	g_warning("%s", g_path_get_basename(filename));
```

Comment:
nah, I think we should just do context detection -- i.e. read out a u32 at offset 0x0 and if it's `0x04034b50` then treat it as an archive, otherwise treat as raw json.

Reply:
I don't think that's even needed.  Just try archive first and then try it as json.  If it fails then bail.  I've adjusted.